File: server/models/Result.js
const mongoose = require('mongoose');

const resultSchema = new mongoose.Schema({
  studentName: {
    type: String,
    required: [true, 'Student name is required'],
    trim: true
  },
  courseName: {
    type: String,
    required: [true, 'Course name is required'],
    trim: true
  },
  examName: {
    type: String,
    required: [true, 'Exam name is required'],
    trim: true
  },
  percentage: {
    type: Number,
    required: [true, 'Percentage is required']
  },
  grade: {
    type: String,
    required: [true, 'Grade is required'],
    trim: true
  },
  status: {
    type: String,
    enum: ['active', 'inactive'],
    default: 'active'
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Result', resultSchema);

File: mobile/lib/core/models/result.dart
class Result {
  final String id;
  final String courseName;
  final String examName;
  final double percentage;
  final String grade;

  Result({
    required this.id,
    required this.courseName,
    required this.examName,
    required this.percentage,
    required this.grade,
  });

  factory Result.fromJson(Map<String, dynamic> json) => Result(
        id: json['_id'] ?? json['id'] ?? '',
        courseName: json['courseName'] ?? '',
        examName: json['examName'] ?? '',
        percentage: (json['percentage'] ?? 0).toDouble(),
        grade: json['grade'] ?? 'F',
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'courseName': courseName,
        'examName': examName,
        'percentage': percentage,
        'grade': grade,
      };
}
